pallets/gear/src/ext.rs

```diff
@@ -265,10 +259,8 @@ impl EnvExt for LazyPagesExt {
                 continue;
             }
             self.fresh_allocations.insert(wasm_page);
-            wasm_page.to_gear_pages_iter().for_each(|page| {
-                log::trace!("add {:?} to lazy pages", page);
-                save_page_lazy_info(id, page);
-            });
+            let gear_pages: BTreeSet<PageNumber> = wasm_page.to_gear_pages_iter().collect();
```

Comment:
I would provide also into_gear_pages_iter

And also I don't see any reason to collect iter to convert it into it afterward